### PR TITLE
Closes #2315: Downgrade menu_block to 8.x-1.7 until we have a better solution for empty sidebars getting rendered.

### LINKED ISSUE #2315: Empty Sidebar First region rendered on pages not in the menu
## Problem/Motivation
While testing Quickstart 2.6.1 on the [Test environment for the Think Tank site](https://test-azs-thinktank.pantheonsite.io/), we noticed that the sidebar region is appearing on all pages of the test site, whether those pages are in the sidebar menu or not.

### Describe the bug
In this case of the Think Tank site, the sidebar is using a custom menu instead of Main Navigation. For pages which are not in the sidebar menu, the HTML for the sidebar region is included but is empty:

```html
<div class="sidebar_first sidebar col-md-3 order-first" id="sidebar_first">
  <aside class="section" role="complementary">
                  
  </aside>
</div>
```

## Proposed resolution
This problem may be related to this Drupal issue: [Empty blocks get displayed](https://www.drupal.org/project/menu_block/issues/3280764). We could try applying the patch from this issue to fix the problem with the empty Sidebar First region.

However, this patch cannot be applied currently because recently we added a patch from another Drupal issue to Quickstart: [Menu block renders when tree is empty as of 8.x-1.8 [regression]](https://www.drupal.org/project/menu_block/issues/3271218).

@trackleft noted that patch 17 from this issue removes the use statement for AccountInterface:
`-use Drupal\Core\Session\AccountInterface;`

Since the patch from the first linked issue requires the use of AccountInterface for menu blocks, @trackleft suggested removing the other patch from Quickstart so that the "Empty blocks get displayed" patch could be applied.

### Expected behavior

The sidebar region should not appear if the region is empty (no sidebar menu blocks are displayed).

##  Screenshots
image

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Downgrades menu_block back to 8.x-1.7 (the version used in Quickstart 2.5.x).  This should resolve the immediate problems identified in issue #2315 for now for sites upgrading to Quickstart 2.6.x and give us time to find a better solution to deal with the larger core issue that is the underlying cause of the issue: https://www.drupal.org/project/drupal/issues/953034

This reverts PR #2049.

Verified that no database or config schema changes exist between 8.x-1.7 and 8.x-1.10.

## Related issues
Closes #2315 

## How to test
- Add a custom menu
- Add a menu block for the new menu to the sidebar first region
- Verify that an empty sidebar is not displayed for pages not in the menu

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [x] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -180,9 +180,6 @@
             "drupal/media_migration": {
                 "Catch migrate RequirementsExceptions (3268480)": "https://www.drupal.org/files/issues/2022-03-08/3268480-2.patch"
             },
-            "drupal/menu_block": {
-                "Menu block renders when tree is empty as of 8.x-1.8 [regression] (3271218)": "https://www.drupal.org/files/issues/2022-04-29/menu_block_rendered_empty-3271218-17.patch"
-            },
             "drupal/migrate_plus": {
```
[the remark this answers is not in the document]
Comment (trackleft) [reply composer.json:183]: It does not

**Comment by trackleft** [comment]
I've tested this with both system menu blocks and custom menu blocks.   In 2.5.5 custom menu blocks cause the ghost block to appear, however , the system menu blocks function as expected.

**Review by trackleft — APPROVED** [review]

**Comment by trackleft** [comment]
Other notes:

The patch we are using is likely superceded by [issue 3007225 on drupal.org](https://www.drupal.org/project/menu_block/issues/3007225) 
Menu block 1.10, combined with the patch in [ issue 3280764 on drupal.org](https://www.drupal.org/project/menu_block/issues/3280764) does not work with any version of PHP less than 8.1

This patch in [issue 3271218 on drupal.org](https://www.drupal.org/node/3271218) does not work with menu_block 1.10

**Review by bberndt-uaz — APPROVED** [review]
Tested in Probo just in case: I created a new system menu, added the menu as a block to the Sidebar First region, and disabled the default Quickstart sidebar menu block. With the Initial visibility level set to 1, the new sidebar block appears on all pages. With the Initial visibility level set to 2, the sidebar properly appears only on [the page which is included in the new menu](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2320.probo.build/pages/card-decks).

**Pull request merged**